---
title: ""
format: html
---

# Genomic data wrangling and processing

Your data is now on a server/cloud/local computer, ready for analysis. Now what?


## Check that the data are not corrupt

Believe it or not, computers can make mistakes! When your data files are being copied over from the sequencing facility to cloud storage, errors can occur that cause your files to become corrupted. This is not always obvious based on file size or a quick look at the contents. The first thing you'll want to do is run a checksum, which will output a fixed-length message of 16 bytes-- a unique identifier of each file. The files should also come with the checksums run by the technician at the sequencing facility. Check out the section on [Data Integrity in our GA shell workshop](https://genomicsaotearoa.github.io/shell-for-bioinformatics/2_download_data/) for more information on how to run this. 


## Genomic file types

The files you will get back from the sequencing facility will most likely end in `.fastq.gz`

The `.gz` indicates it is a compressed gzip file. For most genomic analyses, you do not need to unzip these first--this saves you storage space! 

The `fastq` indicates the file is in [FASTQ format](https://en.wikipedia.org/wiki/FASTQ_format). FASTQ files are a text-based format that stores the raw read sequences, along with a quality score for each base. 

The quality score for illumina comes from the intensity of the fluorescent signal during sequencing by synthesis. For nanopore, requires an additional step to convert the electrical current change into the equivalent format? can be trickier? 

[INSERT PICTURE OF FASTQ FILES - mayeb spotty data as real example?]


-  fastq, gz, r1 and r2 files  



- quality encoding  
- fastqc in depth  (use to teach intro shell )

### Other genomic file types 

- intro to data types and words you'll hear:
    - VCF files
    - BAM files  
  - gtfs    

  - fa (shorthand?)


## Good vs bad genomes

Somewhat colloquially, you will hear geneticists refer to "good" genome assemblies and "bad" genome assemblies.But how do you know if the species you are working on has a good or bad genome, if it has one at all?

There are a few metrics we use

N50
busco score 
contig level, scaffold level, chromosome level. 

Show two exmaples screenshot from NCBI of good and bad. (and excellent?)

There also can be a good assembly - highly contiguos sequence (once the reads are assembled into bigger strings we call them contigs), as clsoe to actual chromosoems as possible
and with it may or may not have good annotations - this is actually describing where all the genomic features are, e.g., gene start and end points, intron and exon coordinates, promoter regions, which genes are protein-coding, pseudogenes, snoRNA, tRNA, CDS, gene_id ; transcript_id ; db_xref 
   


- NOT trimming   - can be covered in respective data type workshop. 
