<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.31">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>data-wrangling-and-processing ‚Äì Genomic Data Carpentry (Aotearoa edition)</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./data-storage-and-management.html" rel="prev">
<link href="./theme-images/GA-image-black-favicon.png" rel="icon" type="image/png">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-e1a5c8363afafaef2c763b6775fbf3ca.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-709f788519238433456b39eacb8aeed0.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
<meta property="og:title" content="‚Äì Genomic Data Carpentry (Aotearoa edition)">
<meta property="og:description" content="">
<meta property="og:image" content="figures/fastq-reads.png">
<meta property="og:site_name" content="Genomic Data Carpentry (Aotearoa edition)">
</head>

<body class="nav-sidebar docked quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./data-wrangling-and-processing.html">Genomic data wrangling and processing</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Genomic Data Carpentry (Aotearoa edition)</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/GenomicsAotearoa/genomic-data-carpentry-nz" title="Github" class="quarto-navigation-tool px-1" aria-label="Github"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./test-styling.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">_test styles</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./outtakes-notes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">_outtakes/notes</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./organisation-tidiness.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Organisation and tidy data</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./planning-for-submission.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Planning for submission</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./data-storage-and-management.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Genomic data storage and management</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./data-wrangling-and-processing.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Genomic data wrangling and processing</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#genomic-data-wrangling-and-processing" id="toc-genomic-data-wrangling-and-processing" class="nav-link active" data-scroll-target="#genomic-data-wrangling-and-processing">Genomic data wrangling and processing</a>
  <ul>
  <li><a href="#command-line-interface-cli-vs-graphical-user-interface-gui" id="toc-command-line-interface-cli-vs-graphical-user-interface-gui" class="nav-link" data-scroll-target="#command-line-interface-cli-vs-graphical-user-interface-gui">Command-line interface (CLI) vs graphical user interface (GUI)</a></li>
  <li><a href="#first-check-that-the-data-are-not-corrupt" id="toc-first-check-that-the-data-are-not-corrupt" class="nav-link" data-scroll-target="#first-check-that-the-data-are-not-corrupt">First, check that the data are not corrupt</a></li>
  <li><a href="#understanding-genomic-file-types" id="toc-understanding-genomic-file-types" class="nav-link" data-scroll-target="#understanding-genomic-file-types">Understanding genomic file types</a>
  <ul class="collapse">
  <li><a href="#other-genomic-file-types" id="toc-other-genomic-file-types" class="nav-link" data-scroll-target="#other-genomic-file-types">Other genomic file types</a></li>
  </ul></li>
  <li><a href="#fastqc---quality-scoring-of-raw-reads" id="toc-fastqc---quality-scoring-of-raw-reads" class="nav-link" data-scroll-target="#fastqc---quality-scoring-of-raw-reads">FASTQC - quality scoring of raw reads</a></li>
  <li><a href="#good-genome-bad-genome" id="toc-good-genome-bad-genome" class="nav-link" data-scroll-target="#good-genome-bad-genome">Good genome, bad genome?</a>
  <ul class="collapse">
  <li><a href="#exercises-6-mins" id="toc-exercises-6-mins" class="nav-link" data-scroll-target="#exercises-6-mins">EXERCISES üß†üèãÔ∏è‚Äç‚ôÄÔ∏è (6 mins):</a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content column-body" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block"></header>
<div class="workshop-banner">
  WORK IN PROGRESS
</div>




<section id="genomic-data-wrangling-and-processing" class="level1">
<h1>Genomic data wrangling and processing</h1>
<p>Your data is now on a server/cloud/local computer, ready for analysis. Now what?</p>
<section id="command-line-interface-cli-vs-graphical-user-interface-gui" class="level2">
<h2 class="anchored" data-anchor-id="command-line-interface-cli-vs-graphical-user-interface-gui">Command-line interface (CLI) vs graphical user interface (GUI)</h2>
<p>Command line interface (CLI) and graphic user interface (GUI) are different ways of interacting with a computer‚Äôs operating system. They have different pros and cons. Most people are familiar with the GUI as it is the default interface for most software, particularly on Windows and Mac OS. When using the GUI, you see and interact with visual representations of files, folders, applications, and most other functions of your computer. When using the CLI, you work largely with text representations of software, files, folders, input and output. The shell is a program that allows you to control your computer by typing instructions on the CLI with a keyboard.</p>
<p>There are several reasons to learn how to use the CLI:</p>
<ul>
<li>For most bioinformatics tools, there are no graphical interfaces. <strong>If you want to work in metagenomics or genomics, you‚Äôre going to need to use the CLI/ shell.</strong><br>
</li>
<li>The shell gives you power and allows you to work more efficiently. Tasks that are repetitive (e.g.&nbsp;renaming hundreds of files) can be automated. Tasks that are tedious (e.g.&nbsp;testing a range of input parameters) can be simplified.<br>
</li>
<li>To use remote computers or cloud computing, you often need to use the shell.</li>
</ul>
<p>Here we will show you a quick introduction to using the shell to quality check your raw reads. For more on the shell, see our workshops on <a href="https://genomicsaotearoa.github.io/BioinformaticsTrainingProgramme/portfolio.html#introduction-to-shell-for-bioinformatics">Introduction to Shell</a> and <a href="https://genomicsaotearoa.github.io/BioinformaticsTrainingProgramme/portfolio.html#intermediate-shell-for-bioinformatics">Intermediate Shell</a>.</p>
<blockquote class="blockquote">
<p>Note: you‚Äôll hear the terms ‚Äòshell‚Äô, ‚Äòbash‚Äô, ‚Äòunix‚Äô, ‚Äòterminal‚Äô and ‚Äòcommand-line‚Äô used almost interchangeably. For beginners, you can think of them as essentially all the same thing! As you gain more computational skills, you can dig into the differences.</p>
</blockquote>
</section>
<section id="first-check-that-the-data-are-not-corrupt" class="level2">
<h2 class="anchored" data-anchor-id="first-check-that-the-data-are-not-corrupt">First, check that the data are not corrupt</h2>
<p>Believe it or not, computers can make mistakes! When your data files are being copied over from the sequencing facility to cloud storage, errors can occur that cause your files to become corrupted. This is not always obvious based on file size or a quick look at the contents. The first thing you‚Äôll want to do is run a checksum, which will output a fixed-length message of 16 bytes‚Äì a unique identifier of each file. The files should also come with the checksums run by the technician at the sequencing facility. Check out the section on <a href="https://genomicsaotearoa.github.io/shell-for-bioinformatics/2_download_data/">Data Integrity in our GA shell workshop</a> for more information on how to run this.</p>
</section>
<section id="understanding-genomic-file-types" class="level2">
<h2 class="anchored" data-anchor-id="understanding-genomic-file-types">Understanding genomic file types</h2>
<p>The files you will get back from the sequencing facility will most likely end in <code>.fastq.gz</code></p>
<p>The <code>.gz</code> indicates it is a compressed gzip file. For most genomic analyses, you do not need to unzip these first‚Äìthis saves you storage space, as unzipped files can be double or triple the zipped size!</p>
<p>The <code>fastq</code> indicates the file is in <a href="https://en.wikipedia.org/wiki/FASTQ_format">FASTQ format</a>. FASTQ files are a text-based format that stores the raw read sequences, along with a quality score for each base.</p>
<p>The first 4 lines of a fastq file describe one sequence read:</p>
<p><img src="figures/fastq-reads.png" class="img-fluid"></p>
<p>The <strong>first line is the header information</strong>, beginning always with an <code style="color:black">@</code>. This is the instrument-specific, run identifier and read identifiying information.</p>
<ul>
<li><code style="color:black">V350304715</code> = instrument-specific<br>
</li>
<li><code style="color:black">L3</code> = lane 3 on the flow cell. Can be used for troubleshooting if something globally went wrong with sequencing.<br>
</li>
<li><code style="color:black">C001</code> = cluster identifier (can be used for troubleshooting, same as above).<br>
</li>
<li><code style="color:black">R001</code> = Read 1. All the reads in this file are read 1. If you have paired-end data, you‚Äôll have a corresponding read 2 file.<br>
</li>
<li><code style="color:black">00020454</code> = unique identifier for that read.</li>
<li><code style="color:black">/1</code> = indicates this is a pair member. Only paired-end data have the <code style="color:black">/1</code> or <code style="color:black">/2</code> additional identifier.</li>
</ul>
<p>Each instrument and chemistry will have slightly different header information. The main thing to look out for are indications of the data being read 1 or read 2 (indicates antisense or sense strand data, for protocols that can differentiate strandedness) and whether it is part of a mate pair (indicated by the <code style="color:black">/1</code> or <code style="color:black">/2</code>)</p>
<p>The <strong>second line is the actual sequence</strong>. Count the number of bases ‚Äìwhich read length was chosen for this dataset?</p>
<p>The <strong>third line is just a ‚Äú+‚Äù</strong>. This can be optionally followed by the same sequence identifier (and any description) again.</p>
<p>The <strong>fourth line is the quality score</strong>. Each base in line 2 has a corresponding quality score, indicated by an ASCII character in line 4.</p>
<p>From left to right, these are the quality scores in increasing order of quality:</p>
<pre><code>!"#$%&amp;'()*+,-./0123456789:;&lt;=&gt;?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz{|}~</code></pre>
<p>So, if a single base has a quality score of <code style="color:black">!</code>, which corresponds to a Q-score of 0, this indicates exceptionally poor quality. The sequencing machine was not able to resolve which base it should be. Often these will be shown as N‚Äôs.</p>
<p>Conversely, if a single base has a quality score of <code style="color:black">I</code>, which corresponds to a Q-score of 40, this indicates very high quality. The sequencing machine was ‚Äúvery confident‚Äù that the base called is correct (Q40 indicates an error rate of 0.0001, i.e., a 1 in 10,000 chance of the base being called wrong. <code style="color:black">I</code> is near the upper end of typically achieved quality values, but higher is possible.</p>
<p>For more on quality scores, you can also read this <a href="https://help.basespace.illumina.com/files-used-by-basespace/quality-scores">Illumina guide here</a>. Note there are different types of encoding, but you will most frequently encounter <a href="https://en.wikipedia.org/wiki/Phred_quality_score">Phred</a>+33.</p>
<p>We can assess the quality score across all reads using software called <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a>, and trim down reads to remove poorer quality bases below a set threshold (generally with newer sequence data, you can use Q30 or Q40 as a cut-off). We won‚Äôt cover trimming here, see our workshop on <a href="https://genomicsaotearoa.github.io/BioinformaticsTrainingProgramme/portfolio.html#rna-seq-data-analysis">RNA-seq Data Analysis</a> for a step-by-step guide.</p>
<section id="other-genomic-file-types" class="level3">
<h3 class="anchored" data-anchor-id="other-genomic-file-types">Other genomic file types</h3>
<p>We‚Äôve talked about <code>fastq</code> and <code>gz</code> files types, but there are some other common genomic file types you should know about.</p>
<p>You should also be aware that the extension on the file type is largely there just so you as a human know what kind of file it is. When using the CLI, the computer does not care what the extension says.</p>
<table class="caption-top table">
<thead>
<tr class="header">
<th style="text-align: right;">Extension</th>
<th>Name</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;"><code>.fasta</code> or <code>.fa</code>, <code>.fna</code>, <code>.faa</code></td>
<td>FASTA, FASTA nucleic acid, FASTA amino acid</td>
<td>Contains sequence data. Each sequence starts with a single header line starting with a <code style="color:black">&gt;</code>, followed by the nucleotide (or amino acid in case of <code>.faa</code>) sequence on the second line. No quality encoding included.</td>
</tr>
<tr class="even">
<td style="text-align: right;"><code>.bam</code> <code>.sam</code></td>
<td><strong>B</strong>inary/<strong>S</strong>equence Alignment Map</td>
<td>Contains alignment information of your reads/sequences that are mapped against a reference sequence (e.g., genome). BAM are the compressed, computer-readable only version; SAM are human-readable but larger files. Somewhat interchangeable file types.</td>
</tr>
<tr class="odd">
<td style="text-align: right;"><code>.vcf</code></td>
<td>Variant Call Format</td>
<td>Contains sequence variants called relative to a reference genome. Split into two sections; the header information and the records. See our <a href="https://genomicsaotearoa.github.io/Introduction-to-R/02-data-prelude/#how-variant-calls-are-stored-in-vcf-files">Intro R workshop, section on VCF files</a>.</td>
</tr>
<tr class="even">
<td style="text-align: right;"><code>.gtf</code></td>
<td>General Feature Format</td>
<td>Contains genome annotation data.</td>
</tr>
</tbody>
</table>
<p><br>
</p>
<p><strong>Real examples:</strong></p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-1-contents" aria-controls="callout-1" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
FASTA
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-1" class="callout-1-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>FASTA file containing cds data from a genome, opened in a text editor: <img src="figures/fasta-file.png" class="img-fluid"></p>
</div>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-2-contents" aria-controls="callout-2" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
BAM/SAM
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-2" class="callout-2-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>add screenshots!</p>
</div>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-3-contents" aria-controls="callout-3" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
VCF
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-3" class="callout-3-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>VCF file opened in excel: <img src="figures/vcf-file.png" class="img-fluid"></p>
</div>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-4-contents" aria-controls="callout-4" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
GTF
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-4" class="callout-4-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>GTF file opened in a text editor: <img src="figures/gtf-file.png" class="img-fluid"></p>
</div>
</div>
</div>
</section>
</section>
<section id="fastqc---quality-scoring-of-raw-reads" class="level2">
<h2 class="anchored" data-anchor-id="fastqc---quality-scoring-of-raw-reads">FASTQC - quality scoring of raw reads</h2>
<ul>
<li>fastqc in depth (use to teach intro shell )</li>
</ul>
</section>
<section id="good-genome-bad-genome" class="level2">
<h2 class="anchored" data-anchor-id="good-genome-bad-genome">Good genome, bad genome?</h2>
<p>You will often hear geneticists refer (somewhat colloquially) to <em>‚Äúgood‚Äù</em> and <em>‚Äúbad‚Äù</em> genome assemblies. But what does this actually mean? And how can you tell whether a species you are working on has a good genome assembly?</p>
<p>In practice, we assess genome assemblies using a few metrics that capture <strong>contiguity</strong> and <strong>completeness</strong>.</p>
<p><strong><em>Common genome assembly metrics</em></strong></p>
<ul>
<li><p><strong>Contiguity</strong> is often measured as contig and scaffold N50, which is the length cutoff for the longest contigs that contain 50% of the total genome length. In this era of long-read genome assemblies, a contig N50 over 1 Mb is generally considered good. <a href="https://www.pacb.com/blog/beyond-contiguity/">Excerpt taken from PacBio article.</a></p>
<p>Also see <a href="https://en.wikipedia.org/wiki/N50,_L50,_and_related_statistics">Wiki entry on N50/L50</a>.</p></li>
<li><p><strong>Completeness</strong> is often measured using a BUSCO (<strong>B</strong>enchmarking <strong>U</strong>niversal <strong>S</strong>ingle-<strong>C</strong>opy <strong>O</strong>rthologs) score, which measures completeness using sets of conserved, single-copy genes expected to be present in most species within a specified lineage.</p></li>
</ul>
<p>Reported as percentages of complete, duplicated, fragmented, and missing genes. &lt;- change to example.</p>
<p>Genome assemblies can be described more qualitatively as (in order of less to more complete):</p>
<ul>
<li><strong>Contig-level</strong> ‚Äì sequences are assembled but not ordered or oriented. Often can be thousands to hundreds of thousands of contigs.<br>
</li>
<li><strong>Scaffold-level</strong> ‚Äì contigs linked together, often with gaps. None are placed (i.e., chromosome known) or localised (i.e., orientation on chromosome known).<br>
</li>
<li><strong>Chromosome-level</strong> ‚Äì one or more chromosomes complete, but some unplaced scaffolds may be present.<br>
</li>
<li><strong>Complete genome</strong> - all chromosomes are resolved with no gaps. Not necessarily telomere-to-telomere (T2T).</li>
</ul>
<p>See <a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/glossary/">NCBI Glossary for Assembly level</a>.</p>
<p><br>
</p>
<p><strong><em>What do we mean by a ‚Äúgood‚Äù genome?</em></strong></p>
<p>A <em>good</em> genome assembly is typically one that is:</p>
<ul>
<li><p><strong>Highly contiguous</strong><br>
Reads have been assembled into long sequences called <em>contigs</em>, ideally approaching full chromosome lengths.</p></li>
<li><p><strong>Biologically complete</strong><br>
Most expected genes are present (high BUSCO completeness).</p></li>
<li><p><strong>Structurally accurate</strong><br>
Large-scale errors such as mis-joins or collapsed repeats are minimised.</p></li>
</ul>
<p><br>
</p>
<p><strong><em>Assembly quality vs annotation quality</em></strong></p>
<p>It is important to distinguish between:</p>
<p><strong>Genome assembly</strong> i.e., the reconstructed DNA sequence itself, and the <strong>Genome annotation</strong>, which is the identification and labelling of genomic features, including gene structures (exons, introns, and CDS), non-coding RNAs (e.g., rRNA, tRNA and lncRNA), and associated identifiers such as gene_id, transcript_id, and database cross-references.</p>
<p>A genome can be <strong>well assembled but poorly annotated</strong>, or vice versa.</p>
<section id="exercises-6-mins" class="level3">
<h3 class="anchored" data-anchor-id="exercises-6-mins">EXERCISES üß†üèãÔ∏è‚Äç‚ôÄÔ∏è (6 mins):</h3>
<p><strong><em>How would you describe these genomes? Look at the assembly stats and make some notes of how ‚Äògood‚Äô you think each assembly is.</em></strong></p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
EXERCISE #1 üß†üèãÔ∏è‚Äç‚ôÄÔ∏è (3 mins) - Human Genome Assembly
</div>
</div>
<div class="callout-body-container callout-body">
<p><a href="https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000001405.40/">Human Genome assembly GRCh38.p14</a></p>
<p><img src="figures/genome-assembly-human-01.png" class="img-fluid"> Note: RefSeq is the curated database. GenBank is the non-curated, anyone-can-submit-to database.</p>
<p><img src="figures/genome-assembly-human-02.png" class="img-fluid"></p>
<p>Jot down your thoughts before looking at the solution below:</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-5-contents" aria-controls="callout-5" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Human genome solution
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-5" class="callout-5-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Chromosome level - very high quality. busco score - very close to complete. very good genome</p>
<p>number of extra scaffolds and contigs is very low (470 s / 996 c - it can be thousands of extra so hundreds is pretty good!)</p>
<p>scaffold and contig N50. describe these!</p>
</div>
</div>
</div>
<p><br>
</p>
</div>
</div>
<p><br>
</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
EXERCISE #2 üß†üèãÔ∏è‚Äç‚ôÄÔ∏è (3 mins) - Sea Anemone Genome Assembly
</div>
</div>
<div class="callout-body-container callout-body">
<p><a href="https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000209225.1/"><em>Nematostella vectensis</em> (sea anemone) Genome assembly ASM20922v1</a></p>
<p><img src="figures/genome-assembly-nemvec-01.png" class="img-fluid"></p>
<p><img src="figures/genome-assembly-nemvec-02.png" class="img-fluid"></p>
<p>Jot down your thoughts before looking at the solution below:</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-8-contents" aria-controls="callout-8" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Sea anemone solution
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-8" class="callout-8-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>scaffold level - medium quality. but the N50 is below 1mb, at 472.6kb scaffold. not great! quite fragmented.</p>
<p>busco score pretty good despite lack of placed scaffolds. contig/scaffold N50 - high enough that most genes are resolved.</p>
<p>assembly itself not amazing, but annotation is probably good enough for gene level compariosn with other species.</p>
<p>Larger scale structural variants and synteny difficult or erroneous to resolve. -&gt; is this true?</p>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-7-contents" aria-controls="callout-7" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Extra for experts
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-7" class="callout-7-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Here is the original paper for the curious: <a href="https://www.science.org/doi/10.1126/science.1139158">Putnam et al.&nbsp;2007.</a></p>
<p>The authors claim the genome reveals <em>‚Äúancestral eumetazoan genomic organization‚Äù</em>. How did they assess this and what conclusion did they make about the eumetazoan ancestor resemblance to modern vertebrates, sea anemones and prokaryotes? Do you think they had a ‚Äògood‚Äô enough genome to assess this?</p>
</div>
</div>
</div>
</div>
</div>
</div>
<p><br>
</p>
</div>
</div>


</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "Óßã";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<nav class="page-navigation column-body">
  <div class="nav-page nav-page-previous">
      <a href="./data-storage-and-management.html" class="pagination-link" aria-label="Genomic data storage and management">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Genomic data storage and management</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
  </div>
</nav>
</div> <!-- /content -->




</body></html>